### PR TITLE
[SPARK-37669][PYTHON] Remove unnecessary usages of OrderedDict

### PR DESCRIPTION
**Pull request by ueshin**
### What changes were proposed in this pull request?

Removes unnecessary usages of `OrderedDict`.

### Why are the changes needed?

Now that supported Python is 3.7 and above, we can remove unnecessary usages of `OrderedDict` because built-in `dict` guarantees the insertion order.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by ueshin** [comment]
cc @HyukjinKwon @itholic @xinrong-databricks @zero323

**Comment by zero323** [comment]
Looks reasonable (there is virtually zero chance we're going to support Python implementation that doesn't use ordered dictionaries in foreseeable future).

File: python/pyspark/pandas/frame.py [hunk 1]
```diff
@@ -9651,7 +9651,7 @@ def stack(self) -> DataFrameOrSeries:
                 ).with_filter(SF.lit(False))
             )
 
-        column_labels: Union[defaultdict, OrderedDict] = defaultdict(dict)
+        column_labels: Dict[Label, Dict[Any, Column]] = defaultdict(dict)
```
Comment (zero323) [inline python/pyspark/pandas/frame.py:9654]: Should we use [`DefaultDict`](https://docs.python.org/3/library/typing.html#typing.DefaultDict) here?

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/pandas/frame.py (hunk 1 above)
Comment (ueshin) [reply python/pyspark/pandas/frame.py:9654]: `column_labels` will be re-assinged at https://github.com/apache/spark/blob/93b21059d26bf1e88ad4101b71d0dae7bf4a6c5c/python/pyspark/pandas/frame.py#L9669
I think it's okay to use more generic type.

Comment (HyukjinKwon) [reply python/pyspark/pandas/frame.py:9654]: Yeah, I think this is fine too.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**